records_mover/db/redshift/loader.py

```diff
@@ -23,11 +23,19 @@ class RedshiftLoader(LoaderFromRecordsDirectory):
     def __init__(self,
                  db: Union[sqlalchemy.engine.Engine, sqlalchemy.engine.Connection],
                  meta: sqlalchemy.MetaData,
-                 temporary_s3_directory_loc: Callable[[], ContextManager[BaseDirectoryUrl]])\
+                 s3_temp_base_loc: Optional[BaseDirectoryUrl])\
```

Comment:
I was doing a too-clever-by-half thing before and passing in a function that creates a temporary directory in the right place in S3.

Now I need to know things like "do I actually have a bucket in which to create the temporary directory", so I'm passing in something more low level (at the price of duplicating a three line function between two classes with no inheritance relationship).